### PR TITLE
llvmPackages_latest: llvm9->llvm10

### LINKED ISSUE #84099: llvmPackages and llvmPackages_latest are outdated
## Update: Current progress

- `llvmPackages`: #85151
- `llvmPackages_latest`: #84096

## Issue description

- `llvmPackages` still points to LLVM 7, while version 10 is already out now.
- `llvmPackages_latest` could/should be bumped to LLVM 10, as #83350 is merged now.

cc:
- Maintainers: @lovek323 @7c6f434c @dtzWill
- Recent PRs: @DieGoldeneEnte @ggreif

Note: This is optional and I'm not familiar with the process here, but to me it seems like it would be a good time to bump both aliases, if someone has time for it (noticed this while reviewing the LLVM 10 backport).

### PR DESCRIPTION
**Pull request by DieGoldeneEnte**
###### Motivation for this change

Since llvm10 is now the newest release.
This also sets llvm-polly and clang-polly to normal llvm/clang, since polly is now enabled by default.

Maybe we want to remove the `-polly` variants at some point in the future?

The only affected packages are:
 * google-cloud-cpp
 * tiledb
 * python37Packages.tiledb/python38Packages.tiledb (doesn't build, see below)

###### Things done
The tiledb python module has tests failing, but this is unrelated to llvm (see #84091 for more information).

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -7994,8 +7994,10 @@ in
       useMacosReexportHack = true;
     };
   };
-  llvm-polly = llvmPackages_latest.llvm-polly;
-  clang-polly = llvmPackages_latest.clang.override { cc = llvmPackages_latest.clang-polly-unwrapped; };
+
+  # llvm and clang are build with polly by default
```
Comment (ggreif) [inline pkgs/top-level/all-packages.nix:7998]:
```suggestion
  # llvm and clang are built with polly by default
```

**Comment by ggreif** [comment]
I guess this should be `git rebase --onto staging`, since it probably causes a lot of rebuilds?

**Comment by DieGoldeneEnte** [comment]
This causes almost no rebuilds, because `llvmPackages_latest` is almost never used in packages.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (DieGoldeneEnte) [reply pkgs/top-level/all-packages.nix]: After looking how this is usually done, I changed the lines to use throw. This prevents `llvm-polly`/`clang-polly`, but also provides an error message to the user. I think this is the best solution.

Thanks for finding the typo :)

**Comment by DieGoldeneEnte** [comment]
@lovek323 @7c6f434c @dtzWill could you take a look?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7994,8 +7994,9 @@ in
       useMacosReexportHack = true;
     };
   };
-  llvm-polly = llvmPackages_latest.llvm-polly;
-  clang-polly = llvmPackages_latest.clang.override { cc = llvmPackages_latest.clang-polly-unwrapped; };
+
+  llvm-polly = throw "clang is now built with polly-plugin by default";
+  clang-polly = throw "clang is now built with polly-plugin by default";
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:7999]: Put this in `pkgs/top-level/aliases.nix` with a timestamp.

**Pull request merged**